api/server.go

```diff
@@ -0,0 +1,299 @@
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/lbryio/lbry.go/errors"
+	"github.com/lbryio/lbry.go/util"
+	"github.com/lbryio/lbry.go/validator"
+	v "github.com/lbryio/ozzo-validation"
+
+	"github.com/fatih/color"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cast"
+)
+
+const authTokenParam = "auth_token"
+
+// Server HTTP Header Settings. Set on header if exists
+// ie. "Content-Type" - "application/json; charset=utf-8"
+// ie. "X-Content-Type-Options" - "nosniff"
+// ie. "X-Frame-Options" - "deny"
+// ie."Content-Security-Policy" - "default-src 'none'"
+// ie. "X-XSS-Protection" - "1; mode=block"
+// ie. "Server" - "lbry.io"
+// ie. "Referrer-Policy" - "same-origin"
+// ie. "Strict-Transport-Security" - "max-age=31536000; preload"
+// ie. "Access-Control-Allow-Origin" -"<header.Origin>"
+// ie. "Access-Control-Allow-Methods" - "GET, POST, OPTIONS"
+var HeaderSettings *map[string]string
+
+// ErrorHandler Allows specific error logging for the server at specific points.
+var ErrorHandler = func(err error) {}
```

Comment:
I would do this to ensure that if someone passes their own error handler function, its correct:

```go
type ErrorHandlerFunc func(error)
var ErrorHandler ErrorHandlerFunc = func(error) {}
```

I'd also rename this to LogErrorFunc and LogError, since that's more accurate. The error doesnt need to be handled (that already happens). It just needs to be logged.

Reply:
golang knows whether or not the function signature matches. If you don't pass the right signature it will not compile. I didn't see another reason to create a type for it. See screen shot below
<img width="882" alt="screen shot 2018-05-12 at 9 09 44 pm" src="https://user-images.githubusercontent.com/3402064/39962836-e49f1d46-5628-11e8-8341-2f3071facf12.png">

I will renamed the var though, makes sense.